### PR TITLE
CEDS-1485 Fix UCR missing in persisted Submission

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala [hunk 1]
```diff
@@ -79,9 +79,11 @@ trait CustomsExportsBaseSpec
   implicit val ec: ExecutionContext = app.injector.instanceOf[ExecutionContext]
 
   implicit lazy val patience: PatienceConfig =
-    PatienceConfig(timeout = 5.seconds, interval = 50.milliseconds) // be more patient than the default
+    PatienceConfig(timeout = 5.seconds, interval = 10.milliseconds)
```
Comment (bambuchaAdm) [inline test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala]: why we should poll more aggressive ?

File: app/uk/gov/hmrc/exports/movements/controllers/ConsolidationController.scala [hunk 2]
```diff
@@ -66,7 +66,7 @@ class ConsolidationController @Inject()(
     )
 
   private def submitMovementConsolidation(
-    actionType: String
+    actionType: ActionType
```
Comment (ed0906) [inline app/uk/gov/hmrc/exports/movements/controllers/ConsolidationController.scala]: 👍

File: test/unit/uk/gov/hmrc/exports/movements/models/SubmissionFactorySpec.scala [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+          actionType = ActionType.ShutMucr,
+          ucrBlocks = Seq(UcrBlock(ucr = ucr_2, ucrType = "M"))
+        )
+
+        compareSubmissions(submission, expectedSubmission)
+      }
+    }
+  }
+
+  private def compareSubmissions(actual: Submission, expected: Submission): Unit = {
```
Comment (ed0906) [inline test/unit/uk/gov/hmrc/exports/movements/models/SubmissionFactorySpec.scala]: Would a `actual must equal(expected)` not suffice?

File: test/utils/PatientCustomsFutures.scala [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * limitations under the License.
+ */
+
+package utils
+
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.duration._
+
+trait PatientCustomsFutures extends ScalaFutures {
```
Comment (bambuchaAdm) [inline test/utils/PatientCustomsFutures.scala]: `IntegrationPatience` - 5s with 1s poll

File: test/unit/uk/gov/hmrc/exports/movements/services/SubmissionServiceSpec.scala [hunk 5]
```diff
@@ -21,35 +21,39 @@ import org.mockito.Mockito.{verify, verifyZeroInteractions, when}
...
+import scala.concurrent.duration._
 import scala.util.control.NoStackTrace
+import uk.gov.hmrc.exports.movements.models.notifications.UcrBlock
+import scala.concurrent.{ExecutionContext, Future}
 
 class SubmissionServiceSpec extends WordSpec with MockitoSugar with ScalaFutures with MustMatchers {
 
   implicit val defaultPatience: PatienceConfig =
-    PatienceConfig(timeout = Span(5, Seconds), interval = Span(100, Millis))
+    PatienceConfig(timeout = 5.seconds, interval = 10.milliseconds)
```
Comment (PatrykRudnicki) [inline test/unit/uk/gov/hmrc/exports/movements/services/SubmissionServiceSpec.scala]: Why 10 millisecond?

File: test/utils/PatientCustomsFutures.scala [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  implicit lazy val patience: PatienceConfig =
```
Comment (PatrykRudnicki) [inline test/utils/PatientCustomsFutures.scala]: I'm not sure about value of this trait and `implicit lazy val patience`. IMO better place for PatienceConfig is base spec.

File: test/unit/uk/gov/hmrc/exports/movements/models/SubmissionFactorySpec.scala (hunk 3 above)
Comment (MaciejRewera) [reply test/unit/uk/gov/hmrc/exports/movements/models/SubmissionFactorySpec.scala]: I don't need to compare all the fields. Some of them, like UUID, or timestamp are not relevant here.

File: test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala (hunk 1 above)
Comment (MaciejRewera) [reply test/unit/uk/gov/hmrc/exports/movements/base/CustomsExportsBaseSpec.scala]: Reverted this change.

File: test/unit/uk/gov/hmrc/exports/movements/services/SubmissionServiceSpec.scala (hunk 5 above)
Comment (MaciejRewera) [reply test/unit/uk/gov/hmrc/exports/movements/services/SubmissionServiceSpec.scala]: Change reverted.

**Pull request merged**